ehrql/file_formats/main.py

```diff
@@ -48,6 +48,30 @@ def read_rows(filename, column_specs, allow_missing_columns=False):
 
 
 def read_tables(filename, table_specs, allow_missing_columns=False):
+    # If we've got a single-table input file and only a single table to read then that's
+    # fine, but it needs slightly special handling
+    if not input_filename_supports_multiple_tables(filename):
```

Comment:
Should this also check if `filename` exists and raise a different error? It could be a non-existent directory, which would return `False` from `filename.is_dir()`, and then you'd get a misleading error if you were trying to read multiple tables

Reply:
Yep, very good point. I'll fix this.